src/renderers/dom/fiber/ReactDOMFiberEntry.js

```diff
@@ -602,6 +619,29 @@ var ReactDOMFiber = {
       // get `true` twice. That's probably fine?
       return true;
     } else {
+      if (__DEV__) {
+        const rootEl = getReactRootElementInContainer(container);
+        const hasNonRootReactChild = !!(rootEl &&
+          ReactDOMComponentTree.getInstanceFromNode(rootEl));
+
+        // Check if the container itself is a React root node.
+        const isContainerReactRoot =
+          container.nodeType === 1 &&
+          isValidContainer(container.parentNode) &&
```

Comment:
@spicyj Should we check for a previous sibling that is a container too? To support this warning properly for comment mount points.